### DISCUSSION TITLE
Can I run minified code through the Google Closure compiler and then apply obfuscation?

### DISCUSSION THREAD

**Discussion by ertugrulcetin** [javascript-obfuscator/javascript-obfuscator#1143] [Q&A] [answered]
As the title says, can I apply obfuscation to minified code (Google Closure)?

It seems that we can not do the opposite;
> Can I run a minifier such as UglifyJS or Google Closure Compiler on the obfuscated output?
No, it's not recommended and in some cases it'll break the code (such as if you enable self-defending). You can run your code through a minifier before to make sure that it removes dead code and do other optimizations, though.

**Comment by slig** [comment, marked as answer]
Yes, any valid JS code can be obfuscated.
